cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/InvalidFieldException.java

```diff
@@ -20,14 +20,15 @@
 import co.cask.cdap.spi.data.table.field.FieldType;
 
 /**
- * Exception thrown when a field is not part of the table schema.
+ * Exception thrown when a field is invalid.
  */
 public class InvalidFieldException extends IllegalArgumentException {
```

Comment:
I made a comment about whether a custom exception should extends from `IllegalArgumentException` or not in https://github.com/cdapio/cdap/pull/10898#discussion_r245493708

In short, I think this exception should not be an `IllegalArgumentException`. `IllegalArgumentException` is for unexpected, programmatic error, while the `InvalidFieldException` is expected to happen.

Reply:
Makes sense, changed it to extend from Exception